### PR TITLE
update development roadmap

### PR DESCRIPTION
**Pull request by jhamman**
- [x] Passes `pre-commit run --all-files`

cc @pydata/xarray

### CODE REVIEW

File: doc/roadmap.rst [hunk 1]
```diff
@@ -181,9 +194,44 @@ development would include:
 -  Possibly moving some infrequently used backends to third-party
    packages.
 
+Flexible higher-level data structures
+-------------------------------------
```
Comment (TomNicholas) [inline doc/roadmap.rst:198]:
```suggestion
-------------------------------------
~~~~~~~~~~~~~~~~
```

File: doc/roadmap.rst [hunk 2]
```diff
@@ -181,9 +194,44 @@ development would include:
...
+often better represented by a collection of related Datasets. Particular common
+examples include:
+
+-  Multi-resolution datasets,
+-  Collections of time series datasets with differing lengths,
+-  Heterogeneous datasets comprising multiple different types of related
+   observational or simulation data,
+-  Bayesian workflows involving various statistical distributions over multiple
+   variables,
+-  Whole netCDF files containing multiple groups.
```
Comment (TomNicholas) [inline doc/roadmap.rst:216]:
```suggestion
-  Whole netCDF files containing multiple groups.
-  Comparison of output from many similar models (such as in the IPCC's Coupled Model Intercomparison Projects)
```

File: doc/roadmap.rst [hunk 3]
```diff
@@ -181,9 +194,44 @@ development would include:
 -  Possibly moving some infrequently used backends to third-party
    packages.
 
+Flexible higher-level data structures
```
Comment (TomNicholas) [inline doc/roadmap.rst:197]: Shouldn't there also be a bit about `xarray.Variable`?

**Review by TomNicholas — COMMENTED** [review]
Thanks @jhamman , I could only think of a couple of small things to add

File: doc/roadmap.rst (hunk 3 above)
Comment (jhamman) [reply doc/roadmap.rst]: added in 0a04faf

**Review by rabernat — APPROVED** [review]

**Review by shoyer — APPROVED** [review]
Thanks Joe!

**Pull request merged**